lib/glific/trackers/tracker.ex

```diff
@@ -0,0 +1,223 @@
+defmodule Glific.Trackers.Tracker do
+  @moduledoc """
+  The tracker object
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  import Ecto.Query, warn: false
+
+  alias __MODULE__
+
+  alias Glific.{
+    Partners.Organization,
+    Partners.Saas,
+    Repo
+  }
+
+  @required_fields [:organization_id, :counts, :date, :period]
+  @optional_fields []
+  @type t() :: %__MODULE__{
+          __meta__: Ecto.Schema.Metadata.t(),
+          id: non_neg_integer | nil,
+          organization_id: non_neg_integer | nil,
+          organization: Organization.t() | Ecto.Association.NotLoaded.t() | nil,
+          counts: map() | %{},
+          date: Date.t() | nil,
+          period: String.t() | nil,
+          inserted_at: :utc_datetime | nil,
+          updated_at: :utc_datetime | nil
+        }
+
+  schema "trackers" do
+    field(:date, :date)
+    field(:period, :string)
+    field(:counts, :map, default: %{})
+
+    belongs_to(:organization, Organization)
+
+    timestamps(type: :utc_datetime)
+  end
+
+  @doc """
+  Standard changeset pattern we use for all data types
+  """
+  @spec changeset(Tracker.t(), map()) :: Ecto.Changeset.t()
+  def changeset(tracker, attrs) do
+    tracker
+    |> cast(attrs, @required_fields ++ @optional_fields)
+    |> validate_required(@required_fields)
+  end
+
+  @doc """
+  Create tracker
+  """
+  @spec create_tracker(map()) :: {:ok, Tracker.t()} | {:error, Ecto.Changeset.t()}
+  def create_tracker(attrs) do
+    %Tracker{}
+    |> Tracker.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  Update tracker
+  """
+  @spec update_tracker(Tracker.t(), map()) ::
+          {:ok, Tracker.t()} | {:error, Ecto.Changeset.t()}
+  def update_tracker(%Tracker{} = tracker, attrs) do
+    tracker
```

Comment:
Maybe I am missing something but I think we don't combine these functions and schema info (struct) in the same file. 

We are putting these functions in context and all the DB, schema-related things to the different file.

Reply:
@pankaj-ag I have made requested changes